.github/workflows/earthly-build.yaml

```diff
@@ -17,57 +17,99 @@ jobs:
         run: |
           sudo wget https://github.com/earthly/earthly/releases/latest/download/earthly-linux-amd64 -O /usr/local/bin/earthly
           sudo chmod 755 /usr/local/bin/earthly
-      - name: Build Space ROS
+      - name: Build Test Space ROS
         id: build
         run: |
-          earthly --ci --output +sources
-          earthly --ci --output +build-testing
-          echo "archivename=$(basename log/build_results_archives/build_results_*.tar.bz2)" >> $GITHUB_ENV
+          earthly --ci --output +build-test --VCS_REF="$(git rev-parse HEAD)"
       - name: Upload build results archive
         uses: actions/upload-artifact@v3
         with:
-          name: ${{env.archivename}}
-          path: log/build_results_archives/${{env.archivename}}
+          name: build_results.tar.bz2
+          path: build_test_results.tar.bz2
           if-no-files-found: error
-  space-ros-image:
+  space-ros-main-image:
     runs-on: ubuntu-latest
+    env:
+      IMAGE_TAG: "latest"
+      VCS_REF: ${{ github.sha }}
+
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: Login to ghcr
+
+      - name: Set up earthly
+        run: |
+          sudo wget https://github.com/earthly/earthly/releases/latest/download/earthly-linux-amd64 -O /usr/local/bin/earthly
+          sudo chmod 755 /usr/local/bin/earthly
+
+      - name: Build spaceros image
+        run: |
+          earthly --ci --output +main-image --VCS_REF="$VCS_REF" --IMAGE_TAG="$IMAGE_TAG" --SKIP_BUILD_TEST=true
+
+      - name: Login to GHCR
         uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Only push the main builds to GHCR
+      - name: Push the main spaceros image to GHCR
+        if: ${{ github.ref_type == 'tag' || github.ref_name == 'main' }}
+        run: |
+          earthly --ci --push +push-main-image --IMAGE_TAG="$IMAGE_TAG" --VCS_REF="$VCS_REF"
+
+      - name: Login to Dockerhub
+        uses: docker/login-action@v3
+        with:
+          username: osrfbot
+          password: ${{ secrets.DOCKER_HUB_RW_TOKEN }}
```

Comment:
Ugh yeah, this is so silly but I noticed that this secret wasn't actually what I thought it was. I think getting it changed might be more trouble than it's worth so probably just easiest to keep this as the silly cli login method:

```
env:
           DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_RW_TOKEN }}
           LATEST_IMAGE_TAG: ${{ env.latest_image_tag }}
         run: |
           echo $DOCKER_HUB_TOKEN | docker login --username osrfbot --password-stdin
```

Reply:
Looks like the CI fails with the new changes I made for this. I was hoping this stdin approach could be skipped. I will revert the changes for the time being.